### PR TITLE
Bugfix: Runtime's Receiver() should only return ID addresses

### PR DESCRIPTION
**Pull request by arajasek**

### CODE REVIEW

**Comment by anorth** [comment]
😮 .

Could we add an assertion or something directly next to the return point of `Receiver()` and `Caller()` that will summon the four horsemen if the address it's about to return is not an ID-address?

**Comment by arajasek** [comment]
Note to self to implement @anorth's suggestion above^ before merging this

File: chain/vm/runtime.go [hunk 1]
```diff
@@ -29,8 +29,30 @@ import (
 	"github.com/filecoin-project/lotus/chain/types"
 )
 
+type Message struct {
+	msg types.Message
+}
+
+func (m *Message) Caller() address.Address {
+	if m.msg.From.Protocol() != address.ID {
+		panic("runtime message has a non-ID caller")
```
Comment (arajasek) [inline chain/vm/runtime.go:38]: Okay to just panic here?

**Comment by arajasek** [comment]
Will need to be merged into the 0.8.0 branch behind an upgrade version check

**Comment by arajasek** [comment]
Good for review, should be merged into `next` after #3936 lands

File: chain/vm/runtime.go (hunk 1 above)
Comment (magik6k) [reply chain/vm/runtime.go:35]: Probably

**Review by magik6k — APPROVED** [review]
This definitely should get some time in a devnet

**Review by Kubuxu — COMMENTED** [review]
High level seems ok

**Pull request merged**